src/platform/nrfconnect/ExternalFlashManager.h

```diff
@@ -31,20 +30,20 @@ class ExternalFlashManager
         SLEEP
     };
 
-    virtual ~ExternalFlashManager() {}
+    // Not copyable or movable
+    ExternalFlashManager(ExternalFlashManager &&)                  = delete;
+    ExternalFlashManager & operator=(const ExternalFlashManager &) = delete;
+    ExternalFlashManager(const ExternalFlashManager &)             = delete;
+    ExternalFlashManager & operator=(ExternalFlashManager &&)      = delete;
 
-    virtual void DoAction(Action aAction)
-    {
-#if defined(CONFIG_PM_DEVICE) && defined(CONFIG_NORDIC_QSPI_NOR)
-        // utilize the QSPI driver sleep power mode
-        const auto * qspi_dev = DEVICE_DT_GET(DT_INST(0, nordic_qspi_nor));
-        if (device_is_ready(qspi_dev))
-        {
-            const auto requestedAction = Action::WAKE_UP == aAction ? PM_DEVICE_ACTION_RESUME : PM_DEVICE_ACTION_SUSPEND;
-            (void) pm_device_action_run(qspi_dev, requestedAction); // not much can be done in case of a failure
-        }
-#endif
-    }
+    static ExternalFlashManager& getInstance() { return sExternalFlashManager; }
```

Comment:
Usually we use UpperCamelCase for methods.
```suggestion
    static ExternalFlashManager& GetInstance() { return sExternalFlashManager; }
```

Reply:
Also alternatively, you could get rid of sExternalFlashManager field this way (but this is a minor and depends on the taste :) ) 
```suggestion
    static ExternalFlashManager& getInstance() { 
      static ExternalFlashManager sExternalFlashManager;
      return sExternalFlashManager; 
      }
```

Reply:
Right, with these suggestions, `.cpp` file is not needed anymore, so I've deleted it as well :) thanks!

Reply:
Note that Mayer's singleton is elegant and thread safe but it usually comes with additional overhead due to locking, which might be good as a matter of fact. So, in essence these two are not exactly the same thing and it's not only a matter of a taste :)